<launch>

    <!-- SYSTEM MODEL -->
    <param name="lf"            type="double"   value="0.125" />
    <param name="lr"            type="double"   value="0.125" />
    <param name="m"             type="double"   value="1.98" />
    <param name="Iz"            type="double"   value="0.03" />
    <param name="Cf"            type="double"   value="68" />
    <param name="Cr"            type="double"   value="71" />
    <param name="mu"            type="double"   value="0.05" />

    <param name="max_vel"       type="double" value="5.0" />
    <param name="min_vel"       type="double" value="0.1" />
    <!-- should be > 1.0 -->
    <param name="init_vx"     type="double" value="0.1"/>
    <param name="trackShape"  type="str"    value="SLAM_shape1"/>

    <node pkg="lpv_mpc" type="plotCarTrajectory.py" name="visualization" output="screen">
        <param name="plotGPS"     type="bool" value="true" />
    </node>

    <!-- SIMULATOR -->
    <node pkg="lpv_mpc" type="vehicleSimulator.py" name="simulator" output="screen">
        <param name="dt"          type="double" value="0.005" />
        <!-- sensor freq update -->
        <param name="gps_freq_update" type="double" value="1000.0" />

        <!-- delay in the simulator -->
        <param name="lowLevelDyn" type="bool" value="false" />
        <param name="delay_df"    type="double" value="0.0" />
        <param name="delay_a"     type="double" value="0.0" />

        <!-- simulator model parameters -->
        <param name="mu"          type="double" value="0.1" />
        <!-- simulator sensor noise -->
        <param name="n_bound_pr"     type="double" value="2.0" />
        <param name="x_std_pr"       type="double" value="0.02" />
        <param name="y_std_pr"       type="double" value="0.01" />
        <param name="psi_std_pr"     type="double" value="0.01" />
        <param name="vx_std_pr"       type="double" value="0.01" />
        <param name="vy_std_pr"       type="double" value="0.04" />
        <param name="psiDot_std_pr"  type="double" value="0.04" />

        <param name="n_bound"     type="double" value="2.0" />
        <param name="x_std"       type="double" value="0.03" />
        <param name="y_std"       type="double" value="0.03" />
        <param name="psi_std"     type="double" value="0.087" />
        <param name="vx_std"       type="double" value="0.03" />
        <param name="vy_std"       type="double" value="0.12" />
        <param name="v_std"       type="double" value="0.15" />
        <param name="ax_std"       type="double" value="0.052" />
        <param name="ay_std"       type="double" value="0.052" />
        <param name="psiDot_std"  type="double" value="0.122" />


        <param name="init_x"    type="double" value="0.0" />
        <param name="init_y"    type="double" value="0.0" />
        <param name="init_yaw"  type="double" value="0.0" />

    </node>

    <node pkg="lpv_mpc" type="plannerMain.py" name="TrajectoryPlanner" output="screen">
        <param name="Planning_Track"   type="int"      value="1" />
        <param name="max_vel"          type="double"   value="5.0" />
        <param name="min_vel"          type="double"   value="0.0" />
        <param name="N"                type="int"      value="20" />
        <param name="Frecuency"        type="int"      value="20" />
        <param name="halfWidth"        type="double"   value="0.5" />
        <param name="Visualization"    type="int"      value="0" />
        <param name="Testing"          type="int"      value="0" />
    </node>



    <!--DYNAMIC OBSERVER-->
    <node pkg="lpv_mpc"              type="stateEstimator.py" name="state_estimator" output="screen">
        <!-- delay in state estimator -->
         <param name="delay_df"   type="double" value="0.0" />
         <param name="delay_a"    type="double" value="0.0" />
         <param name="estimatorON" type="bool"  value="true"/>
         <!-- ekf properties -->
          <param name="thReset"   type="double" value="0.5" />
          <param name="vSwitch"   type="double" value="1.3" />
          <param name="psiSwitch" type="double" value="1.2" />
          <param name="loopRate"  type="double" value="200"	/>
    </node>

    <!-- CONTROL -->
    <node pkg="lpv_mpc" type="controllerMain.py" name="control" output="log">
        <param name="mode"        type="str"     value="simulations" />
        <param name="saveData"    type="bool"    value="true" />
        <param name="N"           type="int"     value="15" />
        <param name="Hz"          type="double"  value="30" />
        <param name="vel_ref"     type="double"  value="3.0"/>
        <param name="vel_th"      type="double"  value="0.0" />
    </node>


</launch>
